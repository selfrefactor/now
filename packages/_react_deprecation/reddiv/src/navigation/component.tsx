import {
  Container,
  Grid,
  Range,
  RangeContainer,
  UplimitContainer,
  } from './styled'

import * as React from 'react'
import { connect } from 'react-redux'

import { rangeChange } from '../root/actions'
import { Uplimit } from './styled'

export class Navigation extends React.Component<Props, {}> {
  constructor(props: Props) {
    super(props)
    this.onRangeChange = this.onRangeChange.bind(this)
  }
  public onRangeChange(e: any) {
    this.props.dispatch(
      rangeChange(Number(e.target.value)),
    )
  }
  public render() {

    return (
      <div>
        {this.props.store.navigationActive &&
          <Container>
            <Grid>

              <RangeContainer>
                <Range
                  min={0}
                  step={10}
                  max={500}
                  onChange={this.onRangeChange}
                  value={this.props.scrollStore.upLimit}
                />
              </RangeContainer>

              <UplimitContainer>
                <Uplimit>
                  {this.props.scrollStore.upLimit}
                </Uplimit>
              </UplimitContainer>
            </Grid>
          </Container>}
      </div>
    )
  }
}

const connectComponent = ({ store, scrollStore }) => ({ store, scrollStore })

export const NavigationWrapped = connect(connectComponent)(Navigation)
