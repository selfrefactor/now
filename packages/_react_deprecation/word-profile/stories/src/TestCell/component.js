import * as React from 'react'
import styled from 'styled-components'
import { delay, defaultTo, repeat, dropLast } from 'rambdax'

export class TestCell extends React.Component{
  constructor(props){
    super(props)
    this.ref = React.createRef()
    this.Container = styled.div`
      font-size: ${ defaultTo('1.5vh', props.fontSize) } !important;
      background: #fafafa;
      color: #111;
      ${ props.evalStyled ? props.evalStyled : '' }
    `
    this.step = 1
    this.TEST_CELL = defaultTo('TEST_CELL', props.localStorageItem)
    this.shouldProceed = !localStorage.getItem(this.TEST_CELL)

    this.state = { testString: repeat('_', props.start).join`` }
    this.work = this.work.bind(this)
    this._setState = this._setState.bind(this)
  }

  componentDidMount(){
    if (this.shouldProceed){
      this.work()
    }
  }

  async work(){
    while (true){
      const okWidth = this.checkElement()
      if (okWidth){
        localStorage.setItem(this.TEST_CELL, this.state.testString.length)

        return window.location.reload(true)
      }

      await delay(50)

      const testString = dropLast(this.step, this.state.testString)

      await this._setState({ testString })
    }
  }

  checkElement(){
    const el = this.ref.current

    return el.scrollWidth <= el.clientWidth
  }

  render(){
    if (!this.shouldProceed) return null

    return (
      <this.Container ref={this.ref}>
        {this.state.testString}
      </this.Container>
    )
  }

  _setState(newState){
    return new Promise(resolve => {
      this.setState(newState, resolve)
    })
  }
}
