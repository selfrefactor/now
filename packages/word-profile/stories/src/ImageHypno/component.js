import * as React from 'react'
import { ImageHypnoSimple } from './simple'
import { ImageHypnoComplex } from './complex'
const CHUNK_SIZE = 100

export function ImageHypno(props){
  if (props.hypnoList.length < 4 * CHUNK_SIZE){
    console.log('LESS')

    return <ImageHypnoSimple { ...props } />
  }
  console.log('MORE')

  return <ImageHypnoComplex chunkSize={ CHUNK_SIZE } { ...props } />
}
