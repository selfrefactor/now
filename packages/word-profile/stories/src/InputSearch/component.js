import * as React from 'react'
import { library } from '@fortawesome/fontawesome-svg-core'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faIgloo } from '@fortawesome/free-solid-svg-icons'
library.add(faIgloo)

const inputStyle = { paddingLeft : '11%' }

export class InputSearch extends React.Component{
  constructor(props){
    super(props)
    this.handlePress = this.handlePress.bind(this)
  }

  handlePress(e){
    if (e.key === 'Enter'){
      this.props.onSubmit(e.target.value)
    }
  }

  render(){
    return (
      <div>
        <input
          style={ inputStyle }
          onKeyPress={ this.handlePress }
          placeholder={ this.props.placeholder }
        />

        <FontAwesomeIcon icon="igloo" />
      </div>
    )
  }
}

