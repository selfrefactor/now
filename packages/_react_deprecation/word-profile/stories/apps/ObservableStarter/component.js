import React from 'react'
import { connect } from 'react-redux'
import { Grid, Cell } from '../../src/Grid/component'
import { increment } from './actions'

const contentCellProps = {
  evalStyled: 'outline: 1px solid grey',
  topLeft:{
    x:1,
    y:2,
  },
  height:28,
  width: 14,
}

const sidebarCellProps = {
  ...contentCellProps,
  evalStyled: 'outline: 1px solid green',
  topLeft:{
    x:16,
    y:1,
  },
}

const counterStep = 6

export class ObservableStarterComponent extends React.Component{
  constructor(props){
    super(props)
    this.increment = this.increment.bind(this)
  }

  increment(){
    this.props.dispatch(increment(counterStep))
  }

  render(){
    return (
      <Grid>
        <Cell {...contentCellProps}>
          <h2>{this.props.counter}</h2>
        </Cell>
        <Cell {...sidebarCellProps}>
          <div>
            <button type='button' onClick={this.increment}>+</button>
          </div>
        </Cell>
      </Grid>
    )
  }
}

const mapStateToProps = state => ({ counter: state.reducer })

export default connect(mapStateToProps)(ObservableStarterComponent)

