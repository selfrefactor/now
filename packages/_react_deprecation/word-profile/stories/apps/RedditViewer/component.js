import { playBee } from '../bees/play'
import { updateDbBee } from '../bees/updateDb'
import React, { useReducer } from 'react'
import { nextIndex, once } from 'rambdax'
import { takeArguments } from 'string-fn'
import { parseUrlInputsBee } from '../bees/parseUrlInputs'
import { Grid, Cell } from '../../src/Grid/component'
import { VideoStretched } from '../../src/VideoStretched/component'
import { ImageStretched } from '../../src/ImageStretched/component'
import {
  getCurrentState,
  setCurrentState,
  createReducer,
  dispatchEvent,
  componentDidMountRaw,
} from '../../libs/reduxed/index.js'

const {
  subreddit,
  play,
  specialMode,
} = parseUrlInputsBee(
  takeArguments(
    window.location.href,
    undefined,
    true
  )
)
document.title = subreddit

const initialState = {
  db      : [],
  index   : 0,
  marker  : undefined,
  loading : false,
  play,
  subreddit,
}

function incIndexAnt(state){
  const index = nextIndex(state.index, state.db)
  const isGif = state.db[ index ].type === 'gif'

  return {
    ...state,
    index,
    loading : isGif,
  }
}

function rootReducer(state, action){
  switch (action.type){
  case 'SET_DB':
    return {
      ...state,
      marker : action.payload.marker,
      db     : [ ...state.db, ...action.payload.db ],
    }
  case 'GIF_READY':
    return {
      ...state,
      loading : false,
    }
  case 'INC_INDEX':
    return incIndexAnt(state)
  default:
    return state
  }
}

const reducer = createReducer(
  rootReducer,
  getCurrentState,
  {},
  { logFlag: false }
)

const componentDidMount = once(async dispatch => {
  componentDidMountRaw(dispatch)
  await updateDbBee(subreddit)
  playBee(getCurrentState)
})

let Holder
let prevStore

function allowRenderBee(prevStore, store){
  if (store.index !== prevStore.index) return true

  return false
}

function shouldRender(store){
  if (prevStore === undefined){
    prevStore = store

    return false
  }
  const result = allowRenderBee(prevStore, store)
  prevStore = store

  return result
}

export function RedditViewer(){
  const [ store, dispatch ] = useReducer(
    reducer,
    getCurrentState(initialState)
  )
  setCurrentState(store)
  componentDidMount(dispatch)
  if (store.db.length === 0) return null

  const currentInstance = store.db[ store.index ]
  const isImage = !specialMode && currentInstance.type === 'image'
  const isGif = !specialMode && currentInstance.type === 'gif'
  const isVideo = !specialMode && currentInstance.type === 'video'
  if (shouldRender(store) || Holder === undefined){
    Holder =
      (<Grid>
        <Cell
          height={ 32 }
          topLeft={ {
            x : 0,
            y : 0,
          } }
          width={ 32 }
        >
          {isImage && (
            <ImageStretched src={currentInstance.src} />
          )}

          {isGif && (
            <ImageStretched
              callback={dispatchEvent('GIF_READY')}
              src={currentInstance.src}
            />
          )}

          {isVideo && (
            <VideoStretched {...currentInstance} />
          )}
        </Cell>
      </Grid>)

  }

  return Holder
}
