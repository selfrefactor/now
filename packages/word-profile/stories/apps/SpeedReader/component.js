import { getData } from './getData.js'
import {
  defaultTo,
  delay,
  last,
  maybe,
  range,
  toDecimal,
} from 'rambdax'
import React from 'react'
import { initialGetLocalize, setLocalize } from 'client-helpers'
import { Grid, Cell } from '../../src/Grid/component'
import { Options } from '../../src/Options/component'
import { produceRow } from './produceRow.js'
import { getReloadIndexes } from './getReloadIndexes.js'
import styled from 'styled-components'

const BACKGROUND = '#ede8e1aa'

const Div = styled.div`
z-index:1000;
font-size: 6vh;
color: #30322ef1;
margin-top: auto;
margin-bottom: auto;
`
const MarkedDivLeft = styled(Div)`
color: #f7f2f2;
padding-left: 10%;
min-height:7.2vh;
background: #977d96e9;
width: 100%;
`
const MarkedDivRight = styled(MarkedDivLeft)`
color: #f7f2f2;
padding-left: 0%;
background: #977d96e9;
width: 100%;
`

function defineDelay(word, base){
  const longDelay = word.length > 12
  const mediumDelay = word.length > 7
  if (longDelay) return Math.floor(260 * base)
  if (mediumDelay) return Math.floor(220 * base)

  return Math.floor(170 * base)
}

const bookIndexOption = {
  label: 'book.index',
  type: 'SELECT',
  visibleLabel: 'Book index',
  choices: [ '0', '1', '2', '3','4','5','6','7','8','9','10','11' ],
  value: initialGetLocalize({
    key:'book.index',
    defaultValue: '1',
  }),
}
const forceReloadOption = {
  label: 'force.reload',
  type: 'TOGGLE',
  visibleLabel: 'Force reload on every 1% progress',
  value: initialGetLocalize({
    key:'force.reload',
    defaultValue: false,
  }),
}
const progressOption = {
  visibleLabel: 'Progress',
  label: 'speed.reader.progress',
  type: 'SLIDER',
  between: [ 0, 100 ],
  value: initialGetLocalize({
    key:'speed.reader.progress',
    defaultValue: 0,
  }),
}
const speedOption = {
  visibleLabel: 'Speed',
  label: 'speed.reader.speed',
  type: 'SLIDER',
  between: [ 200, 1000 ],
  value: initialGetLocalize({
    key:'speed.reader.speed',
    defaultValue: 500,
  }),
}

function calculateActualSpeed(input){
  return toDecimal(input / 500, 1)
}

const isFocused = x => [ 10, 11 ].includes(x)

export class SpeedReader extends React.Component{
  constructor(props){
    super(props)
    this.SPEED = calculateActualSpeed(speedOption.value)
    this.OPTIONS = [
      bookIndexOption,
      progressOption,
      speedOption,
      forceReloadOption,
    ]
    this.state = {
      show: true,
      word : defaultTo('', props.testString),
    }
    this.work = this.work.bind(this)
    this.handleOptionsCallback = this.handleOptionsCallback.bind(this)
  }

  async work(data){
    const len = data.length
    const reloadIndexes = getReloadIndexes(len)
    const initialCounter = progressOption.value === 0 ?
      -1 :
      reloadIndexes[ progressOption.value ]

    const counter = initialCounter ?
      initialCounter :
      last(reloadIndexes)

    for (const i of range(counter + 1, len)){
      if (reloadIndexes.includes(i)){
        const percentage = reloadIndexes.indexOf(i) + 1
        setLocalize(progressOption.label, percentage)

        if (forceReloadOption.value && percentage < 98){
          return window.location.reload(false)
        }
      }

      if (this.state.show){
        this.setState({ word : data[ i ] })
      }
      await delay(defineDelay(data[ i ], this.SPEED))
    }

    setLocalize(progressOption.label, 0)
    await delay(2000)
    window.location.reload(false)
  }

  handleOptionsCallback({ type }){
    if (type === 'UPDATE_ACTIVE'){
      return this.setState({ show: false })
    }
    if (type !== 'UPDATE_OPTIONS') return
    delay(500).then(() => window.location.reload(false))
  }

  componentDidMount(){
    if (this.props.testString) return

    getData(bookIndexOption.value).then(
      data => this.work(data)
    )
  }

  render(){
    const row = produceRow(this.state.word)

    const Row = row.split``.map((char, i) => {
      const focused = isFocused(i)
      if (!focused && (!char || !this.state.show)) return

      const El = maybe(
        i === 10,
        MarkedDivLeft,
        i === 11 ? MarkedDivRight : Div
      )

      return (
        <Cell
          evalStyled='display:flex;'
          height={ 10 }
          key={ `speed-reader-row-${ i }` }
          topLeft={ {
            x : 3 + i,
            y : 10,
          } }
          width={ 1 }
        >
          <El>{char}</El>
        </Cell>
      )
    })

    return (
      <Grid
        background={BACKGROUND}
      >
        <Options
          callback={this.handleOptionsCallback}
          keyBinding={this.props.optionsKeyBinding}
          label='speed.reader'
          options={this.OPTIONS}
        />
        <Cell
          evalStyled='background: ${BACKGROUND}'
          height={ 24 }
          topLeft={ {
            x : 2,
            y : 4,
          } }
          width={ 28 }
        />
        {this.state.show ? Row : null}

      </Grid>
    )
  }
}
