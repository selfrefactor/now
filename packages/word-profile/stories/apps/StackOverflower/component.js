import { delay } from 'rambdax'
import React from 'react'
import { connect } from 'react-redux'
import { Grid, Cell } from '../../src/Grid/component'
import { Options } from '../../src/Options/component'
import { nextFromUser } from './actions'
import { autoOption, speedOption, tagsOption, batchSizeOption } from './options'

export class StackOverflowerComponent extends React.Component{
  constructor(props){
    super(props)
    this.OPTIONS = [
      tagsOption,
      autoOption,
      speedOption,
      batchSizeOption,
    ]
    this.handleOptionsChange = this.handleOptionsChange.bind(this)
    this.intervalHolder = null
    this.nextFromUser = this.nextFromUser.bind(this)
  }

  handleOptionsChange({ type }){
    if (type !== 'UPDATE_OPTIONS') return
    delay(500).then(() => window.location.reload(false))
  }

  nextFromUser(){
    this.props.dispatch(nextFromUser())
  }

  componentDidMount(){
    if (!this.props.production) return
    const child = document.createElement('div')
    const childSecond = document.createElement('div')
    child.id = 'portal'
    childSecond.id = 'portal-second'
    document.body.appendChild(child)
    document.body.appendChild(childSecond)

    this.props.dispatch({ type: 'INIT' })
    const { dispatch } = this.props

    if (autoOption.value){
      this.intervalHolder = setInterval(() => {
        dispatch({ type: 'NEXT_FROM_INTERVAL' })
      }, speedOption.value * 1000)
    }

  }

  componentWillUnmount(){
    if (!this.intervalHolder) return
    this.intervalHolder = null
  }

  render(){
    const { ready, currentIndex, data } = this.props.store
    if (!ready) return null
    const currentInstance = data[ currentIndex ]
    const { link, accepted_answer_id } = currentInstance

    const answer = `${ link }/${ accepted_answer_id }#${ accepted_answer_id }`

    const toRender = () => (
      <React.Fragment>
        <Cell
          evalStyled='width:100%;z-index:1000;'
          height={2}
          topLeft={{
            x: 11,
            y: 0,
          }}
          width={2}
        >
          <div className='overflow__button' onClick={this.nextFromUser}>
            <div>Next</div>
          </div>

        </Cell>

        <Cell
          evalStyled='width:100%;z-index:1000;'
          height={2}
          topLeft={{
            x: 23,
            y: 0,
          }}
          width={4}
        >
          <div className='button'>
            <div>
              <a href={answer} target='blank'>
              Link
              </a>
            </div>
          </div>
        </Cell>

      </React.Fragment>
    )

    return (
      <Grid>
        <Options
          callback={this.handleOptionsChange}
          keyBinding={this.props.optionsKeyBinding}
          label='stack.overflower'
          options={this.OPTIONS}
        />
        {toRender()}
      </Grid>
    )
  }
}

const mapStateToProps = state => ({ store: state.reducer })

export default connect(mapStateToProps)(StackOverflowerComponent)

