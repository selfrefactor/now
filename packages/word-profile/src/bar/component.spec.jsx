import React from 'react'
import { shallow } from 'enzyme'
import { Bar } from './component'

test('happy', () => {
  expect(
    shallow(<Bar store={{ counter : 2 }} dispatch={console.log}/>)
  ).toMatchSnapshot()
})
