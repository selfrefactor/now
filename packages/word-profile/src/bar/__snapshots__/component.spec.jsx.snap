// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`happy 1`] = `
<div>
  Count: 
  2
  <button
    onClick={[Function]}
  >
    +
  </button>
</div>
`;
